NebulaPatcher/Patches/Transpilers/PlayerAction_Combat_Transpiler.cs

```diff
@@ -0,0 +1,185 @@
+                    .Advance(-1)
+                    .Set(OpCodes.Call, AccessTools.DeclaredPropertyGetter(typeof(NebulaWorld.Combat.CombatManager),
+                        nameof(NebulaWorld.Combat.CombatManager.PlayerId)))
+                );
+
+            return codeMatcher.InstructionEnumeration();
+        }
+        catch (System.Exception e)
+        {
+            Log.Error("Transpiler PlayerAction_Combat.Shoot failed.");
+            Log.Error(e);
+            return instructions;
+        }
+    }
+
+    [HarmonyTranspiler]
+    [HarmonyPatch(nameof(PlayerAction_Combat.Bombing))]
+    public static IEnumerable<CodeInstruction> Bombing_Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        try
+        {
+            //  Broadcast bombing event to other players by replacing ptr.ApplyConfigs();
+
+            var codeMatcher = new CodeMatcher(instructions)
+                .MatchForward(true, new CodeMatch(OpCodes.Call, AccessTools.Method(typeof(Bomb_Liquid), nameof(Bomb_Liquid.ApplyConfigs))))
+                .Insert(new CodeInstruction(OpCodes.Ldarg_0))
+                .Advance(1)
+                .SetOperandAndAdvance(AccessTools.Method(typeof(PlayerAction_Combat_Transpiler), nameof(SendBomb_Liquid)))
+                .MatchForward(true, new CodeMatch(OpCodes.Call, AccessTools.Method(typeof(Bomb_Explosive), nameof(Bomb_Explosive.ApplyConfigs))))
+                .Insert(new CodeInstruction(OpCodes.Ldarg_0))
+                .Advance(1)
+                .SetOperandAndAdvance(AccessTools.Method(typeof(PlayerAction_Combat_Transpiler), nameof(SendBomb_Explosive)))
+                .MatchForward(true, new CodeMatch(OpCodes.Call, AccessTools.Method(typeof(Bomb_EMCapsule), nameof(Bomb_EMCapsule.ApplyConfigs))))
+                .Insert(new CodeInstruction(OpCodes.Ldarg_0))
+                .Advance(1)
+                .SetOperandAndAdvance(AccessTools.Method(typeof(PlayerAction_Combat_Transpiler), nameof(SendBomb_EMCapsule)));
+
+            return codeMatcher.InstructionEnumeration();
+        }
+        catch (System.Exception e)
+        {
+            Log.Error("Transpiler PlayerAction_Combat.Bombing failed.");
+            Log.Error(e);
+            return instructions;
+        }
+    }
+
+    static void SendBomb_Liquid(ref Bomb_Liquid ptr, PlayerAction_Combat combat)
+    {
+        ptr.ApplyConfigs();
+        if (!Multiplayer.IsActive)
+        {
+            return;
+        }
+
+        var packet = new MechaBombPacket(
+            Multiplayer.Session.LocalPlayer.Id,
+            ptr.nearStarId,
+            in combat.player.uVelocity,
+            in ptr.uVel,
+            in ptr.uAgl,
+            ptr.protoId);
+
+        Multiplayer.Session.Network.SendPacket(packet);
+    }
+
+    static void SendBomb_Explosive(ref Bomb_Explosive ptr, PlayerAction_Combat combat)
+    {
+        ptr.ApplyConfigs();
+        if (!Multiplayer.IsActive)
+        {
+            return;
+        }
```

Comment:
Minor nitpick, could be simplified to:
```suggestion
        if (!Multiplayer.IsActive) return;
```